src/badges.js

```diff
@@ -0,0 +1,953 @@
+					invalidParam.push(`template paramater '${detail}.validations' must be an object with properties min, max, regex and regexExplain`);
+				}
+				if ('validations' in template.details[detail]) {
+					if (
+						'min' in template.details[detail]['validations'] &&
+						(typeof template.details[detail]['validations']['min'] !== 'number' || template.details[detail]['validations']['min'] < 0)
+					) {
+						invalidParam.push(`template paramater '${detail}.validations.min' must not be a number below 0`);
+					}
+					if (
+						'max' in template.details[detail]['validations'] &&
+						(typeof template.details[detail]['validations']['max'] !== 'number' || template.details[detail]['validations']['max'] < 0)
+					) {
+						invalidParam.push(`template paramater '${detail}.validations.max' must not be a number below 0`);
+					}
+					if (
+						'min' in template.details[detail]['validations'] &&
+						'max' in template.details[detail]['validations'] &&
+						template.details[detail]['validations']['min'] > template.details[detail]['validations']['max']
+					) {
+						invalidParam.push(`template paramater '${detail}.validations.min' must be a number lower than '${detail}.validations.max'`);
+					}
+					if (
+						'regex' in template.details[detail]['validations'] &&
+						(typeof template.details[detail]['validations']['regex'] !== 'string' || !template.details[detail]['validations']['regex'])
+					) {
+						invalidParam.push(`template paramater '${detail}.validations.regex' must be a string`);
+					}
+					if ('regex' in template.details[detail]['validations'] && !('regexExplain' in template.details[detail]['validations'])) {
+						invalidParam.push(`template paramater '${detail}.validations' When using regex, please also provide regexExplain`);
+					}
+					if (
+						'regexExplain' in template.details[detail]['validations'] &&
+						(typeof template.details[detail]['validations']['regexExplain'] !== 'string' || !template.details[detail]['validations']['regexExplain'])
+					) {
+						invalidParam.push(`template paramater '${detail}.validations.regexExplain' must be a string`);
+					}
+				}
+				break;
+			case 'parameters':
+				if (!Array.isArray(template.details[detail])) {
+					invalidParam.push(`template paramater '${detail}' must be an array`);
+				} else {
+					template.details[detail].map((parameter, i) => {
+						Object.keys(parameter).forEach((key) => {
+							if (!['name', 'type', 'label', 'description', 'defaultValue', 'validations', 'options'].includes(key)) {
+								invalidParam.push(`template paramater '${detail}[${i}].${key}' is not a valid parameter`);
+							}
+							if (['name', 'type', 'label', 'description', 'defaultValue'].includes(key) && (typeof parameter[key] !== 'string' || !parameter[key])) {
+								invalidParam.push(`template paramater '${detail}[${i}].${key}' must be a string with a value`);
+							}
+							const allowedTypes = ['array', 'text', 'color', 'url', 'integer'];
```

Comment:
Other types seem to be supported:
https://github.com/searchspring/searchspring-core/blob/v4.720.2/httpdocs/assets-v2/react-root/src/models/BadgeTemplate.ts#L78-L88